src/openms/include/OpenMS/FILTERING/ID/IDFilter.h

```diff
@@ -1362,6 +1362,15 @@ namespace OpenMS
 
     /// @name Filter functions for class IdentificationData
     ///@{
+
+    template <typename PredicateType>
+    static void filterObservationMatchesByFunctor(
+        IdentificationData& id_data, PredicateType&& func, bool cleanup_affected = false)
+    {
+      id_data.removeFromSetIf_(id_data.observation_matches_, func);
+      if (cleanup_affected) id_data.cleanup();
```

Comment:
Is false a good default?

Reply:
I am not sure. id_data.cleanup() has 5 optional bools for removing related stuff.
I am not sure if we want to remove the related Observation and the IdentifiedPeptide if this ObsMatch is removed. We need to think about it.